runtime-jit-experimental fails in JIT/opt/InstructionCombining/CompareZero

Example pipeline run: https://dev.azure.com/dnceng-public/public/_build/results?buildId=1215527&view=results
Example console log: https://helixr18s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-dfaeef053c694d96ad/JIT.opt/1/console.5cd0ad9e.log

```
__tmp16_CompareZero.cs:253:11: error: ARM64: expected string not found in input
 //ARM64: {{^ *}} tbz {{x[0-9]+}}, #63, G_M{{[0-9]+}}_IG{{[0-9]+}}{{$}}
```

Test introduced by #120667. cc @jonathandavies-arm

Also cc @dotnet/jit-contrib

--- a/src/tests/JIT/opt/InstructionCombining/CompareZero.csproj
+++ b/src/tests/JIT/opt/InstructionCombining/CompareZero.csproj
@@ -13,5 +13,6 @@
     </Compile>
     <CLRTestEnvironmentVariable Include="DOTNET_TieredCompilation" Value="0" />
     <CLRTestEnvironmentVariable Include="DOTNET_JITMinOpts" Value="0" />
+    <CLRTestEnvironmentVariable Include="DOTNET_JitStressProcedureSplitting" Value="0" />
   </ItemGroup>
 </Project>

